iroh-net/src/lib.rs

```diff
@@ -1,11 +1,102 @@
-//! iroh-net provides connectivity for iroh.
+//! Peer-to-peer connectivity based on QUIC.
 //!
-//! This crate is a collection of tools to establish connectivity between peers.  At
-//! the high level [`Endpoint`] is used to establish a QUIC connection with
-//! authenticated peers, relaying and holepunching support.
+//! iroh-net is a library to establish direct connectivity between peers.  It exposes an
+//! interface to [QUIC] connections and streams to the user, while implementing direct
+//! connectivity using [hole punching] complemented by relay servers under the hood.
 //!
-//! The "relay-only" feature forces all traffic to send over the relays. We still
-//! receive traffic over udp and relay. This feature should only be used for testing.
+//!
+//! # Connection Establishment
+//!
+//! An iroh-net connection between two iroh-net nodes is usually established with the help
+//! of a Relay server.  When creating the [`Endpoint`] it connects to the closest Relay
+//! server and designates this as the *home relay*.  When other nodes want to connect they
+//! first establish connection via this home relay.  As soon as connection between the two
+//! nodes is established they will attempt to create a direct connection, using [hole
+//! punching] if needed.  Once the direct connection is established the relay server is no
+//! longer involved in the connection.
+//!
+//! If one of the iroh-net nodes can be reached directly, connectivity can also be
+//! established without involving a Relay server.  The [`NodeAddr`] can also contain a
+//! number of [socket addresses] on which a node is reachable and will be used to establish a
+//! connection.
+//!
+//!
+//! # Encryption
+//!
+//! The connection is encrypted using TLS, like standard QUIC connections.  Unlike standard
+//! QUIC there is no client, server or server TLS key.  Instead each iroh-net node has a
```

Comment:
Would this change work?
```suggestion
//! QUIC there is no client, server or server TLS key and certificate chain.  Instead each iroh-net node has a
```